lib/queue.js

```diff
@@ -277,35 +277,21 @@ Queue.prototype._getJobs = function (queueType, start, end, cb) {
             this.client.zrange(this.toKey(queueType), start, end, idsPromise.defer());
             break;
         }
-      });
 
-    promise = idsPromise.then((ids) => {
-      return Promise.all([...ids].map((id) => this.getJob(id)));
-    });
+        return idsPromise;
+      })
+      .then((ids) => {
+        ids = Array.from(ids);
```

Comment:
What does this accomplish? Is `ids` not a normal array? Please leave a comment explaining that.